### ISSUE TITLE
check shapes in arithmetic operations

### ISSUE THREAD

**Issue by devcrocod** [Kotlin/multik#59]
Add `require` of two shapes to each arithmetic operation between the two ndarrays.
Shapes must match exactly.
Now the check is only for the number of elements.

**Comment by plastic-karma** [comment]
Does this apply to `and`, `or`, `max` and `min` as well ?

**Comment by devcrocod** [comment]
Yes, it can be applied to these functions as well. I now want to release a faster version 0.1.1 with bug fixes, so I'll add a check myself.
thanks
